clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -295,6 +295,47 @@ LogicalResult IntAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DataMemberAttr definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+DataMemberAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                       mlir::cir::DataMemberType ty,
+                       std::optional<size_t> memberIndex) {
+  auto clsStructTy = ty.getClsTy();
+  if (!clsStructTy) {
+    emitError() << "class type of a DataMemberAttr must be a struct type";
```

Comment:
My understanding from this PR is that a struct type is required when you create a `DataMemberType` so this will never be executed? Either add a testcase that exercises this or remove it.

Reply:
Updated. Originally the code is

```cpp
auto clsStructTy = ty.getClsTy().dyn_cast<StructType>();
if (!clsStructTy) {
  // ...
}
```

I just forgot to remove the `if` block after I removed the `dyn_cast` call.